backend/app/services/knowledge/code_wiki/publisher.py

```
@@ -0,0 +1,311 @@
+        The refs still outstanding.
+    """
+    spec = (knowledge_base.json or {}).get("spec", {})
+    pending = [str(ref) for ref in (spec.get(PENDING_INDEX_CLEANUP_KEY) or [])]
+    if not pending:
+        return ()
+
+    still_pending: list[str] = []
+    for doc_ref in pending:
+        try:
+            effects.delete_rag_document(int(doc_ref))
+        except Exception as exc:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)